### DISCUSSION TITLE
Failed to run

### DISCUSSION THREAD

**Discussion by chivas03** [Beormund/Tasmota32-Multi-Zone-Heating-Controller#19] [Q&A]
Dear All,

thank you for the effort in developing the Multi heat controller :-) This is something I was looking for *gg*. As now is summer on our side I am investigating to improve our heating system control to be well prepared for the winter.

I try to run the heating-controller on a ESP32, unfortunately after uploading the files I get following error message:
"BRY: failed to run compiled code 'syntax_error' - /heating.be:26: strict: redefinition of builtin 'format'"

Today I am unfamiliar with Berry therefore I wanted to ask if somebody had similar experience and can support me.

I really appreciate any help you can provide.
Marco

**Comment by Beormund** [comment]
@chivas03 Tasmota berry has a new global 'format' function (an alias for string.format) which was throwing a compile time re-definition error. This has now been fixed. Please download the latest version of hc_host.tapp and install on the ESP32 file system. Thanks for flagging up.

**Reply by chivas03** [reply]
Thanks for the update and information.
Still got the message:
BRY: failed to run compiled code 'syntax_error' - /hc_host.tapp#/heating.be:944: 'Leds' undeclared (first use in this function)

**Reply by Beormund** [reply]
Which firmware are you using? The Leds class is normally in tasmota32 and its variants. See https://tasmota.github.io/docs/Berry_Addressable-LED/

**Reply by chivas03** [reply]
Thanks for the information. I used a tasmota32 build with tasmocompiler. However, I updated to the latest online release and it works now! I don't know why the WS2812 function was integrated with tasmocompiler - maybe my mistake.

I can now investigate to your heat controller :-)
